test/TextForEvent-test.ts

```diff
@@ -0,0 +1,110 @@
+import './skinned-sdk';
+
+import { textForEvent } from "../src/TextForEvent";
+import { MatrixEvent } from "matrix-js-sdk";
+import SettingsStore from "../src/settings/SettingsStore";
+import { SettingLevel } from "../src/settings/SettingLevel";
+import renderer from 'react-test-renderer';
+
+function mockPinnedEvent(
+    pinnedMessageIds?: string[],
+    prevPinnedMessageIds?: string[],
+): MatrixEvent {
+    return new MatrixEvent({
+        type: "m.room.pinned_events",
+        state_key: "",
+        sender: "@foo:example.com",
+        content: {
+            pinned: pinnedMessageIds,
+        },
+        prev_content: {
+            pinned: prevPinnedMessageIds,
+        },
+    });
+}
+
+describe("TextForPinnedEvent - newly pinned message(s)", () => {
```

Comment:
There should be one top level `describe('TextForEvent')`, followed by a nested `describe('textForPinnedEvent')`, then all the `it('should...')` tests.

Reply:
Agree, this is much better, thanks for the suggestion.